listener.go

```diff
@@ -133,31 +133,14 @@ func (l *listener) Close() error {
 	l.closed = true
 	err := l.Listener.Close()
 
-	// Drain nextConn
-drainNextConn:
-	for {
-		select {
-		case conn := <-l.nextConn:
-			conn.Close()
-		default:
-			break drainNextConn
-		}
-	}
 	close(l.nextConn)
+	for conn := range l.nextConn {
+		conn.Close()
+	}
 
-	// Drain nextErr
-drainNextErr:
-	for {
-		select {
-		case e := <-l.nextErr:
-			if e == nil { // it may already be closed
-				break drainNextErr
-			} else {
-				err = e
-			}
-		default:
-			break drainNextErr
-		}
+	// l.nextErr is always closed at this point
+	for e := range l.nextErr {
+		err = e
```

Comment:
I'm not sure if draining the error channel is desired. Why should `Close` return an error that is not caused by the method itself?

Reply:
Ah you're right, I thought this was preserving behavior but I must have been misreading.  Also, it probably needs a more careful look at whether there is anything subtle that was requiring these to be drained together.